json-string.c

```diff
@@ -154,24 +154,26 @@ next(char ** p, char * e, bool html)
 
   // Read the rest
   unsigned char tmp;
-  switch(trail_size) {
-    case 3:
+  switch(trail_size) { // @todo this could be replaced by a while loop 
+  case 3:
```

Comment:
please keep the original code layout so we can compare it with the upstream implementation.  We will change the implementation, but track the upstream one.